compiler/passes/scopeResolve.cpp

```diff
@@ -1548,6 +1548,23 @@ static void resolveUnresolvedSymExpr(UnresolvedSymExpr* unresolvedSymExpr,
 
           unresolvedSymExpr->replace(prim_capture_fn);
           prim_capture_fn->insertAtTail(unresolvedSymExpr);
+          //
+          // If we detect that this function reference is within a
+          // c_fnPtrTo() call then we only need a C pointer to the
+          // function, not a full Chapel first-class function (which
+          // can capture variables).  We'll mark this so that function
+          // resolution can distinguish by adding a second sentinel
+          // argument to the PRIM_CAPTURE_FN call ('true',
+          // arbitrarily).  We could alternatively add another
+          // primitive to distinguish the C case, but for most
+          // intents, it behaves identically to PRIM_CAPTURE_FN, so
+          // this seemed like a waste of a primitive...
```

Comment:
I think it would be cleaner to make a primitive but to have most code do the same thing for both PRIM_CAPTURE_FN and PRIM_FN_PTR (or whatever it is called). That way, distinguishing them is clearer/less magical and yet you should be able to re-use as much code as you want.